tests/test_python.py

```diff
@@ -106,13 +107,29 @@ def test_tox(auto_container):
     "container_per_test", CONTAINER_IMAGES_T1, indirect=["container_per_test"]
 )
 @pytest.mark.parametrize("hmodule, port, retry", [("http.server", port1, 10)])
-def test_python_webserver_1(container_per_test, hmodule, port, retry):
+def test_python_webserver_1(
+    container_per_test, container_runtime, hmodule, port, retry
+):
     """Test that the python webserver is able to open a given port"""
 
+    check = ""
     portstatus = False
-
+    vv = None
     t = 0
 
+    if container_runtime.runner_binary == "podman":
+        vv = LOCALHOST.package(container_runtime.runner_binary).version
+        vn = vv.split(".")
+        assert vn[0].isnumeric()
+
+        # container version: old versions needs a containerized check of background processes
+        if int(vn[0]) < 4:
+            check = f"sleep 1; ss --numeric --listening --tcp | grep {port};"
```

Comment:
Please elaborate in more detail why you are doing the following with podman < 4.0. Also, if you need to check for open sockets, please use testinfra for that: https://testinfra.readthedocs.io/en/latest/modules.html?highlight=port#testinfra.modules.socket.Socket
Adding these shell scripts everywhere will turn into a maintenance nightmare in the long run.

Reply:
After many check and attempts for debugging a failing test, I noted that after stared the python http server, in case of podman v.1.x any next testinfra command remains **waiting** for that process to exit after timeout expired, before to go on, even being in detach mode and background. Therefore those testinfra port/process checks will have that server no more running. This case does not happens with the newer versions of podman.